src/kagglehub/http_resolver.py

```diff
@@ -44,9 +44,10 @@ def __call__(self, handle: str, path: Optional[str] = None):
 
                 # Extract all files to this directory.
                 with tarfile.open(archive_file.name) as f:
-                    # Use 'data' filter to prevent dangerous security issues.
```

Comment:
'data' filter was only added to Python 3.8.

But we do want to support Python 3.7 to match Keras compatibility.